app/src/main/java/com/zeapo/pwdstore/git/GitServerConfigActivity.kt

```diff
@@ -0,0 +1,166 @@
+                protocol = when (checkedId) {
+                    R.id.clone_protocol_https -> Protocol.Https
+                    R.id.clone_protocol_ssh -> Protocol.Ssh
+                    else -> protocol
+                }
+            }
+        }
+
+        binding.connectionModeGroup.addOnButtonCheckedListener { _, checkedId, checked ->
+            if (checked) {
+                connectionMode = when (checkedId) {
+                    R.id.connection_mode_ssh -> ConnectionMode.Ssh
+                    R.id.connection_mode_openkeychain -> ConnectionMode.OpenKeychain
+                    R.id.connection_mode_username -> ConnectionMode.Username
+                    else -> connectionMode
+                }
+            }
+        }
+
+        binding.serverUrl.apply {
+            setText(serverUrl)
+            doOnTextChanged { text, _, _, _ ->
+                serverUrl = text.toString()
+            }
+        }
+
+        binding.serverPort.apply {
+            setText(serverPort)
+            doOnTextChanged { text, _, _, _ ->
+                serverPort = text.toString()
+            }
+        }
+
+        binding.serverUser.apply {
+            setText(serverUser)
+            doOnTextChanged { text, _, _, _ ->
+                serverUser = text.toString()
+            }
+        }
+
+        binding.serverPath.apply {
+            setText(serverPath)
+            doOnTextChanged { text, _, _, _ ->
+                serverPath = text.toString()
+            }
+        }
+
+        binding.saveButton.setOnClickListener {
+            if (isClone && PasswordRepository.getRepository(null) == null)
+                PasswordRepository.initialize(this)
+            if (updateHostname()) {
+                settings.edit(true) {
+                    putString("git_remote_protocol", protocol.toString())
+                    putString("git_remote_auth", connectionMode.toString())
+                    putString("git_remote_server", serverUrl)
+                    putString("git_remote_port", serverPort)
+                    putString("git_remote_username", serverUser)
+                    putString("git_remote_location", serverPath)
+                }
+                if (!isClone)
+                    Snackbar.make(binding.root, getString(R.string.git_server_config_save_success), Snackbar.LENGTH_SHORT).show()
+                else
+                    cloneRepository()
+            } else {
+                Snackbar.make(binding.root, getString(R.string.git_server_config_save_failure), Snackbar.LENGTH_LONG).show()
+            }
+        }
+    }
+
+    /**
+     * Clones the repository, the directory exists, deletes it
+     */
+    private fun cloneRepository() {
+        val localDir = requireNotNull(PasswordRepository.getRepositoryDirectory(this))
+
+        // Warn if non-empty folder unless it's a just-initialized store that has just a .git folder
+        if (localDir.exists() && localDir.listFiles()!!.isNotEmpty() &&
```

Comment:
It seems wasteful to call `listFiles()` thrice.

Reply:
33d0540330e3